demo/api/src/products/entities/product.entity.ts

```diff
@@ -130,22 +130,23 @@ export class Product extends BaseEntity<Product, "id"> implements DocumentInterf
 
     @Property({ type: "json" })
     @Field(() => [ProductDiscounts])
-    discounts: ProductDiscounts[] = [];
+    discounts?: ProductDiscounts[] = [];
+
     @Property({ type: "json" })
     @Field(() => [String])
-    articleNumbers: string[] = [];
+    articleNumbers?: string[] = [];
 
     @Property({ type: "json", nullable: true })
     @Field(() => ProductDimensions, { nullable: true })
-    dimensions?: ProductDimensions;
+    dimensions?: ProductDimensions = undefined;
```

Comment:
Why don't we use null? The field will be null in the database. So the correct definition would be `dimensions: ProductDimensions | null = null`. Or did we decide to ignore this inconsistency? I think we've already talked about this a view times...